### ISSUE TITLE
ParenthesesAroundCondition not caught when `==` in condition

### ISSUE THREAD

**Issue by martingordon** [bbatsov/rubocop#2561]
```
$ rubocop -V
0.34.2 (using Parser 2.2.2.6, running on ruby 2.2.1 x86_64-darwin14)
```

Running the test case:

```ruby
if (foo)
end

if (x = 'foo')
end

if (x == 'foo')
end
```

Results in:
```
/Users/martin/Desktop/test.rb:1:4: C: Don't use parentheses around the condition of an if.
if (foo)
   ^^^^^
```

Should ParenthesesAroundCondition be triggered on `if (x == 'foo')` as well? The exception to the rule only applies on safe assignment in condition, which is correctly not triggered on the second `if` statement.

I'm not sure if this requires a separate fix, but I noticed this when Rubocop autocorrected a `Style/SpaceAfterControlKeyword` offense (`if(x == 'foo')`).

**Comment by jonas054** [comment]
This bug is not present in v0.35.0. Looks like the correction was made [here](https://github.com/bbatsov/rubocop/pull/2398/files#diff-a6efce5304ba5bdac9c74dfab6caa2ca). The PR is described as a refactoring, but apparently it included this bugfix as well.
